File: python/src/acedatacloud/resources/face.py
"""Face transformation resources (``/face/*``)."""

from __future__ import annotations

from typing import Any


class Face:
    """Synchronous face transformation client."""

    def __init__(self, transport: Any) -> None:
        self._transport = transport

    def _call(self, action: str, body: dict[str, Any]) -> dict[str, Any]:
        return self._transport.request("POST", f"/face/{action}", json=body)

    def analyze(self, *, image_url: str, **kwargs: Any) -> dict[str, Any]:
        return self._call("analyze", {"image_url": image_url, **kwargs})

    def beautify(self, *, image_url: str, **kwargs: Any) -> dict[str, Any]:
        return self._call("beautify", {"image_url": image_url, **kwargs})

    def change_age(self, *, image_url: str, **kwargs: Any) -> dict[str, Any]:
        return self._call("change-age", {"image_url": image_url, **kwargs})

    def change_gender(self, *, image_url: str, **kwargs: Any) -> dict[str, Any]:
        return self._call("change-gender", {"image_url": image_url, **kwargs})

    def detect_live(self, *, image_url: str, **kwargs: Any) -> dict[str, Any]:
        return self._call("detect-live", {"image_url": image_url, **kwargs})

    def swap(self, *, source_image_url: str, target_image_url: str, **kwargs: Any) -> dict[str, Any]:
        return self._call(
            "swap", {"source_image_url": source_image_url, "target_image_url": target_image_url, **kwargs}
        )

    def cartoon(self, *, image_url: str, **kwargs: Any) -> dict[str, Any]:
        return self._call("cartoon", {"image_url": image_url, **kwargs})


class AsyncFace:
    """Async face transformation client."""

    def __init__(self, transport: Any) -> None:
        self._transport = transport

    async def _call(self, action: str, body: dict[str, Any]) -> dict[str, Any]:
        return await self._transport.request("POST", f"/face/{action}", json=body)

    async def analyze(self, *, image_url: str, **kwargs: Any) -> dict[str, Any]:
        return await self._call("analyze", {"image_url": image_url, **kwargs})

    async def beautify(self, *, image_url: str, **kwargs: Any) -> dict[str, Any]:
        return await self._call("beautify", {"image_url": image_url, **kwargs})

    async def change_age(self, *, image_url: str, **kwargs: Any) -> dict[str, Any]:
        return await self._call("change-age", {"image_url": image_url, **kwargs})

    async def change_gender(self, *, image_url: str, **kwargs: Any) -> dict[str, Any]:
        return await self._call("change-gender", {"image_url": image_url, **kwargs})

    async def detect_live(self, *, image_url: str, **kwargs: Any) -> dict[str, Any]:
        return await self._call("detect-live", {"image_url": image_url, **kwargs})

    async def swap(self, *, source_image_url: str, target_image_url: str, **kwargs: Any) -> dict[str, Any]:
        return await self._call(
            "swap", {"source_image_url": source_image_url, "target_image_url": target_image_url, **kwargs}
        )

    async def cartoon(self, *, image_url: str, **kwargs: Any) -> dict[str, Any]:
        return await self._call("cartoon", {"image_url": image_url, **kwargs})

File: typescript/src/resources/face.ts
/** Face transformation resources (`/face/*`). */

import { Transport } from '../runtime/transport';

export class Face {
  constructor(private transport: Transport) {}

  private call(action: string, body: Record<string, unknown>): Promise<Record<string, unknown>> {
    return this.transport.request('POST', `/face/${action}`, { json: body });
  }

  analyze(opts: { imageUrl: string; [key: string]: unknown }): Promise<Record<string, unknown>> {
    const { imageUrl, ...rest } = opts;
    return this.call('analyze', { image_url: imageUrl, ...rest });
  }

  beautify(opts: { imageUrl: string; [key: string]: unknown }): Promise<Record<string, unknown>> {
    const { imageUrl, ...rest } = opts;
    return this.call('beautify', { image_url: imageUrl, ...rest });
  }

  changeAge(opts: { imageUrl: string; [key: string]: unknown }): Promise<Record<string, unknown>> {
    const { imageUrl, ...rest } = opts;
    return this.call('change-age', { image_url: imageUrl, ...rest });
  }

  changeGender(opts: { imageUrl: string; [key: string]: unknown }): Promise<Record<string, unknown>> {
    const { imageUrl, ...rest } = opts;
    return this.call('change-gender', { image_url: imageUrl, ...rest });
  }

  detectLive(opts: { imageUrl: string; [key: string]: unknown }): Promise<Record<string, unknown>> {
    const { imageUrl, ...rest } = opts;
    return this.call('detect-live', { image_url: imageUrl, ...rest });
  }

  swap(opts: { sourceImageUrl: string; targetImageUrl: string; [key: string]: unknown }): Promise<Record<string, unknown>> {
    const { sourceImageUrl, targetImageUrl, ...rest } = opts;
    return this.call('swap', { source_image_url: sourceImageUrl, target_image_url: targetImageUrl, ...rest });
  }

  cartoon(opts: { imageUrl: string; [key: string]: unknown }): Promise<Record<string, unknown>> {
    const { imageUrl, ...rest } = opts;
    return this.call('cartoon', { image_url: imageUrl, ...rest });
  }
}
